### ISSUE TITLE
No ConversationUpdate event when sideloading or installing via organisation catalog

### ISSUE THREAD

**Issue by Crazedpeanut** [microsoft/BotFramework-Services#213]
Similar to #169 

This issue is only occuring for 1:1 chats 

I am attempting to store the users id, serviceUrl, and aad Object Id so I can proactively reach out to them.

I am not receiving ConversationUpdate events when sideloading the app or installing via the organisations app catalog.

Manifest: https://gist.github.com/Crazedpeanut/ff7b7051746f0d38e1f298aaad387e14

I have tried a few scenarios, each time replacing the bot to ensure it's the first interaction between the bot and user. Also each time I regenerate the manifest App Id just incase their is something happening there.

- Install via App Studio in  Teams for Windows
- Install via App Studio in Teams for Web
- Install via Organisation Catalog
- Install along with tabs
- Install without with tabs
- Install notification only bot
- Install not notification only bot

Each scenario I am monitoring the requests that I receive. It's only until I manually type something in, when I receive two activities, the ConversationUpdate and the message

**Comment by Crazedpeanut** [comment]
Just updated this issue, the issue only occurs for 1:1 conversations

**Comment by dmvtech** [comment]
Thanks @Crazedpeanut 

Just to confirm; by "_replacing the bot_", you mean that you are creating a new bot channels registration/web app bot/AAD app, correct? 

Also; you are _always_ getting the conversationUpdate, but _**only after**_ the first message is sent by the user that has installed the bot?

**Comment by Crazedpeanut** [comment]
@dmvtech correct!

**Comment by Crazedpeanut** [comment]
Any updates on this one?

**Comment by dmvtech** [comment]
Hi @Crazedpeanut Still looking into it. I'll try and have more info in an update for you tomorrow.

**Comment by Crazedpeanut** [comment]
Thanks!

**Comment by dmvtech** [comment]
Sorry for the delay. Is this working for you at any time? Or is it failing every single time? Per @clearab 's comment [here](https://github.com/microsoft/BotFramework-Services/issues/169#issuecomment-575754306), it may not work 100% of the time. But I doubt that is the case if you can't ever get it to work.

I was able to get it to break in the `OnTeamsMembersAddedAsync` method by using the [teams-conversation-bot](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/57.teams-conversation-bot) sample.

My steps were:

- Create new web app bot
- Enable Teams channel in bot channels
- Setup NGROK and point bot endpoint to that
- Use sample `57.teams-conversation-bot`
- Configure `appsettings.json` with bot appid and password
- Configure `manifest.json` to use appid
- Zip up manifest `TeamsAppManifest` files as `manifest.zip`
- Run bot
- Run `app studio->manifest editor->import an existing app->Edit app->Test and distribute`
- Chat with bot in teams
- Hit breakpoint on `OnTeamsMembersAddedAsync`

The one thing I did notice is that in your manifest.json you had different values for `id:` and for `"bots"[{"botId":`, where I had the same (my bots appid). I don't know if that matters, but worth a shot.

Please try with the sample I used, as well as using the appid of the bot for both `id` and `botid`. Let me know how it goes!

**Comment by Crazedpeanut** [comment]
@dmvtech just to confirm, 
In your test, did you add the bot as personal bot, or a team bot?

I'll updating the ids in the manifest

**Comment by Crazedpeanut** [comment]
@dmvtech I tried to use the bot Id as the app Id for the manifest, but with the same result.

**Comment by Crazedpeanut** [comment]
Personal Bot Added
<img width="578" alt="1" src="https://user-images.githubusercontent.com/7700874/83365761-fe76da00-a3ed-11ea-9b1e-16e73e2641bf.PNG">

No Posts from Bot Framework. Ignore the OPTIONS, they are from the Azure Portal
<img width="859" alt="0" src="https://user-images.githubusercontent.com/7700874/83365771-05055180-a3ee-11ea-8f72-96808854537d.PNG">

Disable Notifications
<img width="511" alt="3" src="https://user-images.githubusercontent.com/7700874/83365772-0898d880-a3ee-11ea-96af-d172ff2be01f.PNG">

Post a Message
<img width="577" alt="4" src="https://user-images.githubusercontent.com/7700874/83365778-0df62300-a3ee-11ea-842c-30f237185726.PNG">

Get some Posts!
<img width="451" alt="5" src="https://user-images.githubusercontent.com/7700874/83365780-12224080-a3ee-11ea-9d44-c78e68df0aaa.PNG">

Manifest App Id and Bot Ids are the same
<img width="276" alt="botId0" src="https://user-images.githubusercontent.com/7700874/83365807-48f85680-a3ee-11ea-9f31-f0ab672b1834.PNG">
![botid1](https://user-images.githubusercontent.com/7700874/83365808-4b5ab080-a3ee-11ea-8e1f-c92113eab10b.PNG)

**Comment by Crazedpeanut** [comment]
@dmvtech it's happening every time

**Comment by clearab** [comment]
Hi - I've escalated to the Teams support team (the event is created in their infrastructure), and will follow here as soon as I hear back.

**Comment by Crazedpeanut** [comment]
Thanks guys

**Comment by Crazedpeanut** [comment]
Hi @clearab, haven't heard anything for a while, are we able to ping the Teams team?

**Comment by clearab** [comment]
Nothing yet - I'll escalate. I'm assuming you're still experiencing this?

**Comment by Crazedpeanut** [comment]
Performed the same steps as above, same result

**Comment by NandaCodeBox** [comment]
Even i am facing the same issue. Raised new issue in the "Company Communicator" repository.

If there is any solution for this fatal issue would be really helpful.

Issue URL: 
OfficeDev/microsoft-teams-company-communicator-app#178

Same issue reported:
https://github.com/microsoft/BotFramework-Services/issues/169

**Comment by Crazedpeanut** [comment]
Hey @NandaCodeBox, I am speaking with Teams support. I'll update this issue once we find a solution

**Comment by Subhasish-msft** [comment]
@Crazedpeanut  @NandaCodeBox , Can you please share your bot id and timestamp of the issue?

**Comment by NandaCodeBox** [comment]
"botId": "63666cdf-1bed-4816-931f-d8453578961f",

**Comment by Subhasish-msft** [comment]
@NandaCodeBox , Can you please share the timestamp of the reported issue?

**Comment by NandaCodeBox** [comment]
@Subhasish-msft <https://github.com/Subhasish-msft>,

Please refer to the below time stamps..as you can refer there are only 2
receptenst are receiving messages but i have installed this apps for three
more in addition to the current two users.

[image: image.png]

**Comment by NandaCodeBox** [comment]
@Subhasish-msft,

Please refer to the below time stamps..as you can refer there are only 2
receptenst are receiving messages but i have installed this apps for three
more in addition to the current two users.

![image](https://user-images.githubusercontent.com/17465542/85977192-bac9bd00-b9f9-11ea-8d87-90bce289e273.png)

**Comment by Subhasish-msft** [comment]
@NandaCodeBox , The shared screenshot is for sending message time stamp.  We need sideloading timestamp to investigate the bot event issue.

**Comment by NandaCodeBox** [comment]
it is just a few minutes back exactly on the same day.

**Comment by NandaCodeBox** [comment]
@Subhasish-msft <https://github.com/Subhasish-msft>,  Is there any update
on this?

**Comment by Subhasish-msft** [comment]
@NandaCodeBox  We checked the log about the bot, and it says the bot is disabled by the tenant admin. This permission issue is why the bot cannot get any events. Could you please check whether the admin disabled the bot?

**Comment by NandaCodeBox** [comment]
No, I have set the appropriate permission and setup policies. Please refer below. Also the Apps were successfully displayed in Side Rail in Teams. 

Could you please let know where this configuration need to be checked. Link would be helpful.

Screenshots from https://admin.teams.microsoft.com/

Permission Policies

![image](https://user-images.githubusercontent.com/17465542/86122399-66a40300-baf5-11ea-937c-a8189e57b06f.png)

Setup Policies

![image](https://user-images.githubusercontent.com/17465542/86122566-ab2f9e80-baf5-11ea-91c3-2aa0c514df86.png)

**Comment by NandaCodeBox** [comment]
Hi @Subhasish-msft,

Is there any update ?

**Comment by Subhasish-msft** [comment]
@NandaCodeBox , There are two manifests for Company Communicator. Can you try adding both app Ids to app setup policy. It seems you are getting the exception because of a different AppId.

**Comment by NandaCodeBox** [comment]
@Subhasish-msft, Just to clarify.I have cross checked ones again as per the deployment documentation the author and user manifest holds the same bot id.

**Comment by MatthiasWeibel** [comment]
@clearab - any updates on this from the Teams support team? We are blocked because of this issue.

**Comment by Subhasish-msft** [comment]
@NandaCodeBox , Did you try to add both app Id in allowed app list ?

**Comment by NandaCodeBox** [comment]
@Subhasish-msft  Yes did tht ! are u able to reolicate this issue ? Have you tried adding this app to more than 3 users?

**Comment by jwsheppa** [comment]
@Subhasish-msft I am running into the same issue with our organization. It looks like installing the application using the App Setup Policies in Teams will not register a conversation ID with the bot (Company Communicator).

**Comment by NandaCodeBox** [comment]
@nishandi, Is this issue got resolved. We have been waiting for many days. Please do support.

**Comment by Subhasish-msft** [comment]
@NandaCodeBox  As the log says bot is disabled by tenant admin, Can you please verify the "AllowedTenants" config value of the web app in azure portal. It should be same tenant as your MS Teams.

**Comment by cleemullins** [comment]
@NandaCodeBox Any updates on this? Per @Subhasish-msft this looks like a per-tenant issue. Can you confirm? 

We'll close this out next week (8/10) if we don't hear back.

**Comment by NandaCodeBox** [comment]
@Subhasish-msft <https://github.com/Subhasish-msft>  /  @cleemullins   No
not at all. Actually I have already provided the artifacts which are
required. Currently the tenant is dissolved. No more access to the
resources in the Subscription.

Also there are multiple people facing the same issue. I would request you
not to close this issue.

**Comment by cleemullins** [comment]
@NandaCodeBox Understood. 

@Subhasish-msft  - Can you please advise review this, and recommend a path forward?

**Comment by Wajeed-msft** [comment]
Internal Tracker: [Bug 952901: No ConversationUpdate event when sideloading or installing bot via organization catalog](https://domoreexp.visualstudio.com/MSTeams/_workitems/edit/952901)

**Comment by tacri** [comment]
The fix for this issue is being rolled out, ETA for WW deployment is the end of this week.

**Comment by AnaHencheIB** [comment]
We have the same problem in Iberia (Tracker: 21381623) ... could you tell us if the problem has been solved or how to solve it?  
Thx in advance!

**Comment by Wajeed-msft** [comment]
@AnaHencheIB - This fix was deployed WW on 19 Aug 2020. If you are still facing this issue, please share the Bot Id and Timestamp of installation.

**Comment by NandaCodeBox** [comment]
@clearab @AnaHencheIB Will check this week and update.
